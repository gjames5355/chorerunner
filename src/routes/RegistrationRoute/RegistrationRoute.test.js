import React from 'react';
import ReactDOM from 'react-dom';
import RegistrationRoute from './RegistrationRoute';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

const history = { push: () => [] };

it('Registration Routerenders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <RegistrationRoute history={history} />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders Registration Route as expected', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <RegistrationRoute history={history} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
