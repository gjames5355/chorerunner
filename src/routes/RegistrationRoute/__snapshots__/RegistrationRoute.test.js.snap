// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Registration Route as expected 1`] = `
<section
  className="register-container"
>
  <p
    className="descriptionLine"
  >
    Assign tasks for your kids to earn rewards in this fun app.
  </p>
  <h2>
    Sign up
  </h2>
  <div
    className="box"
  >
    <form
      className="registration"
      name="registration-form"
      onSubmit={[Function]}
    >
      <div
        className="formItem"
      >
        <label
          className="Label"
          htmlFor="registration-name-input"
        >
          Enter your name
          <span
            className="Required"
          >
            *
          </span>
        </label>
        <input
          className="Input formBox"
          id="registration-name-input"
          name="name"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <div
          role="alert"
        >
          
        </div>
      </div>
      <div
        className="formItem"
      >
        <label
          className="Label"
          htmlFor="registration-username-input"
        >
          Choose a username
          <span
            className="Required"
          >
            *
          </span>
        </label>
        <input
          className="Input formBox"
          id="registration-username-input"
          name="username"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <div
          role="alert"
        >
          
        </div>
      </div>
      <div
        className="formItem"
      >
        <label
          className="Label"
          htmlFor="registration-password-input"
        >
          Choose a password
          <span
            className="Required"
          >
            *
          </span>
        </label>
        <input
          className="Input formBox"
          id="registration-password-input"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
      <div
        role="alert"
      />
      <footer
        className="formFooter"
      >
        <button
          className="Button basicBtn"
          type="submit"
        >
          Sign up
        </button>
         
        <br />
      </footer>
    </form>
  </div>
</section>
`;
